src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsBuildManager.cs

```diff
@@ -77,6 +95,18 @@ public object DesignTimeOutputMonikers
         /// </summary>
         public string BuildDesignTimeOutput(string bstrOutputMoniker)
         {
+            if (DesignTimeInputsBuildManagerBridge?.AppliesTo(_unconfiguredProjectServices.Project.Capabilities) == true)
+            {
+                IDesignTimeInputsBuildManagerBridge bridge = DesignTimeInputsBuildManagerBridge.Value;
+
+                return _threadingService.ExecuteSynchronously(async () =>
+                {
+                    await _threadingService.SwitchToUIThread();
+
+                    return await bridge.GetDesignTimeInputXmlAsync(bstrOutputMoniker);
```

Comment:
Rememinder me, why is this async but required on the UI thread?

Reply:
This can be followup, but it looks like there's some enforced UI-threadiness in this case that isn't needed. The rules you have to follow are:

1) If I'm on the UI thread, I have a guarantee that the state of the project doesn't change underneath me
2) I can be freely called on a background thread (switching to the UI thread if required)

Above in BuildDesignTimeOutput/DesignTimeOutputMonikers I'm forced to the UI thread for what appears to be unneeded reasons. I think you can still have above guarantees without the forced switch to the UI thread.

Reply:
This is in response to the same comment as below (https://github.com/dotnet/project-system/pull/5326#discussion_r317933815) regarding switching to the UI thread if necessary, rather than verifying (throwing) if not on the UI thread.